code/to_bsdd.py

```diff
@@ -251,10 +283,96 @@ def generalization(pe):
         return pe
 
 
-type_to_values = {
-    'boolean': ["TRUE","FALSE"],
-    'logical': ["TRUE","FALSE","UNKNOWN"],
-}
+def need_brackets(s):
+    """ Verify if that name should be annotated with brackets in other definitions.
+      The criteria: minium 4 letters (skip 'ABC')."""
+    if len(s) >= 4:
+        return True
+    else:
+        return False
+    # and at least 2 capital letters (skip 'Width' etc.).
+    # if re.findall(r'\b(?:[a-zA-Z]*[A-Z]){2,}[a-zA-Z]*\b', s):
+    #     return True
+    
+
+def list_ancestors(name, full_list, ancestor_list=[]):
+    """ traverse the list of hierarchy to find all the parents of this object """
+    try:
+        ancestor = full_list[name]['Parent']
+    except KeyError:
+        logging.warning('The entity %s not found in the IFC.' % name)
+    if ancestor:
+        if ancestor in full_list:
+            ancestor_list.append(ancestor)
+            list_ancestors(ancestor, full_list, ancestor_list)
+        else:
+            logging.warning('Ancestor %s not found in the IFC.' % ancestor)
+    return ancestor_list
+
+
+# TODO do we even want to filter anything? Why don't we upload all the concepts?
```

Comment:
Original line of thought:

1) From IFC it's not possible to relate from arbitrary instances to bSDD. Only things that can have classifications can be enriched with bSDD concepts. Therefore it also doesn't make a lot of sense to have these concepts in bSDD.

2) Many things are IFC specific modelling idioms and not meaningful constructs. Example: objectified relationships. You'd want to keep this out of bSDD I suppose.

Reply:
Many concepts were missing, that's why it was broadened, but relations indeed are not desired (already removed). Anything else that comes to your mind? I will push those changes as we need to keep going with translations. Thanks for insights!